### DISCUSSION TITLE
It is needed to have Parabolic on MacOS

### DISCUSSION THREAD

**Discussion by adrianjpv1** [NickvisionApps/Parabolic#751] [General]
I miss this awesome app. I ditched Windows and moved to MacOS, the latter has made me happier in stability, but I miss the native parabolic app I had on Windows. Are not plans to build a MacOS version of this amazing project?

**Comment by nlogozzo** [comment]
Unfortunately, not at the moment no.

I have no Mac systems, nor do i have the funds to get one, so I have no way of developing and testing Mac apps.

**Comment by nlogozzo** [comment]
Update: I got a Mac and Parabolic will soon be coming to macOS as well!!

**Reply by dhrandy** [reply]
That is awesome. Can't wait!

**Comment by adrianjpv1** [comment]
Have u analyzed the possibility of using Hackintosh? If your system
supports it, then, you can perfectly dual boot using it!. I'm currently
writing this email using it. Actually on Sonoma.

**Reply by nlogozzo** [reply]
No, a Mac Mini M2 went on sale for $479 here...so I bought that haha

**Comment by adrianjpv1** [comment]
Nice. I also bought it for my sister. Hope it could help u to work on
porting it to mac! Great job!

**Comment by nlogozzo** [comment]
I no longer have access to a Mac, however Parabolic does not have a QT version that should in theory build and run on macOS without issues...I will try to set that up in the near future.

**Reply by formlessdao** [reply]
> I no longer have access to a Mac, however Parabolic does not have a QT version that should in theory build and run on macOS without issues...I will try to set that up in the near future.

1. Will Aria2 be available by default enabled in the new version you're currently working on that you ported to C++? 
Default is way too slow right now.
2. Also will there be any option to choose between different audio tracks that exist for a particular video just like we can currently choose video resolution?

**Reply by nlogozzo** [reply]
> Will Aria2 be available by default enabled in the new version you're currently working on that you ported to C++?
Default is way too slow right now.

aria2 will be available but will not be enabled by default. Must be enabled in preferences. Same behavior as the current version. 

> Also will there be any option to choose between different audio tracks that exist for a particular video just like we can currently choose video resolution?

Audio tracks of different languages will be able to be selected, yes.

**Reply by formlessdao** [reply]
>  aria2 will be available but will not be enabled by default. Must be enabled in preferences. Same behavior as the current version.

I get that but can you please work on displaying progress of current download while using Aria 2? Why avoid adding this rudimentary functionality?


> Audio tracks of different languages will be able to be selected, yes.

I'm not talking about different languages but the ability to choose between tracks at different compression levels just like we can choose video resolution.

**Reply by nlogozzo** [reply]
> I get that but can you please work on displaying progress of current download while using Aria 2? Why avoid adding this rudimentary functionality?

This is a limitation of yt-dlp. yt-dlp does not support reading progress from aria2c. Therefore, we cannot do anything about it until it is implemented upstream.

> I'm not talking about different languages but the ability to choose between tracks at different compression levels just like we can choose video resolution.

I will look into this.

**Comment by formlessdao** [comment]
> This is a limitation of yt-dlp. yt-dlp does not support reading progress from aria2c. Therefore, we cannot do anything about it until it is implemented upstream.

Persepolis download manager and Motrix both use under the hood Aria2c and both of them displayed progress of a downloaded file just fine.

**Reply by nlogozzo** [reply]
> Persepolis download manager and Motrix both use under the hood Aria2c and both of them displayed progress of a downloaded file just fine. 

We don't use aria2c under the hood. We use yt-dlp under the hood. yt-dlp is the one that controls and manages aria2c if users turn on that option. 

But at the end of the day, Parabolic talks to yt-dlp and yt-dlp only. So if yt-dlp doesn't support aria2c progress reporting, neither do we. 

It's an issue you'll have to open with yt-dlp

**Comment by nlogozzo** [comment]
The app is available on macports: https://ports.macports.org/port/parabolic/

**Reply by j-lakeman** [reply]
Could it be distributed through Homebrew as well pls?
Awesome application btw! Keep up the good work!

**Reply by nlogozzo** [reply]
> Could it be distributed through Homebrew as well pls?

I don't have a Mac and the macPorts distribution is not maintained by me. You, or anyone with a Mac, is more than welcome to create a package for homebrew.

> Awesome application btw! Keep up the good work!

Thank you :)

**Comment by angelussun** [comment]
Where is the macOS app？

**Reply by nlogozzo** [reply]
https://ports.macports.org/port/parabolic/
